### PR TITLE
Update test_Saver__ugrid.py

### PR DESCRIPTION
**Pull request by pp-mo**
Correct comment, since after #6004 the behaviour, and the testcase changed but the comment did not !

@ESadek-MO tiny fix : hope this makes sense to you.

### CODE REVIEW

File: lib/iris/tests/unit/fileformats/netcdf/saver/test_Saver__ugrid.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def mesh_location_size(mesh, location):
 
 
 # A simple "standard" test mesh for multiple uses, which we can use for cubes
-# that *share* a mesh (since we don't support mesh equality).
+# that ** a mesh (since we don't support mesh equality).
```
Comment (ESadek-MO) [inline lib/iris/tests/unit/fileformats/netcdf/saver/test_Saver__ugrid.py:181]: Not sure what this is meant to mean?

"... we can use for cubes that ** a mesh..."

**Review by ESadek-MO — CHANGES_REQUESTED** [review]

File: lib/iris/tests/unit/fileformats/netcdf/saver/test_Saver__ugrid.py (hunk 1 above)
Comment (pp-mo) [reply lib/iris/tests/unit/fileformats/netcdf/saver/test_Saver__ugrid.py:181]: Good spot -- definitely a mistake ! 🤭

**Comment by pp-mo** [comment]
N.B. I just realised this PR uses a branch of the main repo, instead of my fork.
That's probably just a mistake, but no big deal.  I will remove the unwanted branch afterwards.

**Review by ESadek-MO — APPROVED** [review]
LGTM! Thanks @pp-mo

**Pull request merged**